### ISSUE TITLE
Compiler throwing "Too Many Wrappers" error on two-way edges in Graphql schemas

### ISSUE THREAD

**Issue by thormuller** [AlecAivazis/houdini#90]
I've recently updated my schema to be closer to that in the Example. When running the compiler one of my mutations started generating an error: "Too Many Wrappers" (full error at the bottom of this post). I confirmed (after some experimental surgery on the schema) that the issue is the two-way edge between Recipe and User types. Is there any reason the following schema should not work? 

```
type Recipe {
	id: ID!
	title: String!
	myTitle: String
	description: String
	domain: String
	created: Float
	updated: Float
	user(filter: UserFilter): User! @hasInverse(field: recipes)
	archived: Boolean!
	webRecipeID: String
	recipeStructure: String
	totalTime: String
	img: String
	klImages: String
}

type User {
	id: ID!
	displayName: String
	email: String!
	emailVerified: Boolean
	loginProvider: String
	photoURL: String
	created: Float
	lastSignIn: Float
	recipes(filter: RecipeFilter, order: RecipeOrder, first: Int, offset: Int): [Recipe] @hasInverse(field: user)
	recipesAggregate(filter: RecipeFilter): RecipeAggregateResult
}
```

Once I removed the recipes reference on User the error cleared up. Here's the working schema.

```
type Recipe {
	id: ID!
	title: String!
	myTitle: String
	description: String
	domain: String
	created: Float
	updated: Float
	user(filter: UserFilter): User!
	archived: Boolean!
	webRecipeID: String
	recipeStructure: String
	totalTime: String
	img: String
	klImages: String
}

type User {
	id: ID!
	displayName: String
	email: String!
	emailVerified: Boolean
	loginProvider: String
	photoURL: String
	created: Float
	lastSignIn: Float
}
```

In case, it's helpful, here's the Type definition that was directly spawning the error (I didn't get an error when running standard queries).

```
type Mutation {
	addRecipe(input: [AddRecipeInput!]!): AddRecipePayload
}
```

And the mutation:

```	
const addRecipe = mutation<AddRecipe>(graphql`
		mutation AddRecipe($input: AddRecipeInput!) {
			addRecipe(input: [$input]) {
				recipe {
					id
				}
			}
		}
	`);
```

Here's the error:

```
Error: Too many wrappers: undefined
    at inputType (file:///<PATH>/node_modules/houdini/build/cmd.js:58505:15)
    at file:///<PATH>/node_modules/houdini/build/cmd.js:58526:34
    at Array.map (<anonymous>)
    at inputType (file:///<PATH>/node_modules/houdini/build/cmd.js:58523:78)
    at file:///<PATH>/node_modules/houdini/build/cmd.js:58526:34
    at Array.map (<anonymous>)
    at inputType (file:///<PATH>/node_modules/houdini/build/cmd.js:58523:78)
    at file:///<PATH>/node_modules/houdini/build/cmd.js:58457:137
    at Array.map (<anonymous>)
    at file:///<PATH>/node_modules/houdini/build/cmd.js:58455:185
```

**Comment by pixelmund** [comment]
I ran into the same issue while trying to use an string array in my input.

**Comment by AlecAivazis** [comment]
Sorry you guys are running into this. I have been very busy this past week so I haven't had the chance to dig deeply. Will try to find some time this week if someone else hasn't gotten to it by then

**Comment by thormuller** [comment]
BTW, the error is thrown after the runtime artifacts are generated and in my case doesn't seem to effect the actual operation of the code (at least in my tests so far). So it's currently an annoyance but  _not_ blocking.

**Comment by AlecAivazis** [comment]
@thormuller im trying to hunt this down but the schema you have at the top is incomplete, I dont see the definition for `mutation` or the input. Mind updating it?

**Comment by AlecAivazis** [comment]
This should now be fixed in version `0.8.0`
